From 69ca21920fe5bd139cffa8e30eb896e84a0c886e Mon Sep 17 00:00:00 2001
From: Josh Wu <josh.wu@atmel.com>
Date: Thu, 20 Dec 2012 18:40:39 +0800
Subject: [PATCH 262/319] atmel_nand: nfc: According to the data sheet, for
 READID nand command, it is not need to wait for the
 ready/busy pin.

currently, after we issued READID command then wait for the ready/busy pin, sometimes it meet time out. Actually for READID command, we don't need wait for ready/busy pin.i
so this patch will not wait r/b after READID command.

And for message of time out, it should be an error message.
---
 drivers/mtd/nand/atmel_nand_nfc.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mtd/nand/atmel_nand_nfc.c b/drivers/mtd/nand/atmel_nand_nfc.c
index ea42d0e..9e06ee2 100644
--- a/drivers/mtd/nand/atmel_nand_nfc.c
+++ b/drivers/mtd/nand/atmel_nand_nfc.c
@@ -28,7 +28,7 @@ static int nfc_wait_interrupt(struct atmel_nand_host *host, u32 flag)
 	timeout = wait_for_completion_timeout(&host->comp_nfc,
 			msecs_to_jiffies(100));
 	if (timeout == 0) {
-		dev_dbg(host->dev, "interrupt time out????\n");
+		dev_err(host->dev, "interrupt time out???? flag is 0x%08x\n", flag);
 		return -ETIMEDOUT;
 	}
 
@@ -259,6 +259,7 @@ static void nfc_nand_command(struct mtd_info *mtd, unsigned int command,
 	case NAND_CMD_DEPLETE1:
 	case NAND_CMD_RNDOUT:
 	case NAND_CMD_SEQIN:
+	case NAND_CMD_READID:
 		return;
 
 	case NAND_CMD_STATUS_ERROR:
-- 
1.7.10.4

